homeassistant/components/ring/__init__.py

```diff
@@ -69,13 +59,20 @@ def listen_credentials_updater(token: dict[str, Any]) -> None:
             data={**entry.data, CONF_LISTEN_CREDENTIALS: token},
         )
 
-    user_agent, hardware_id = await get_auth_agent_id(hass)
+    if CONF_DEVICE_ID not in entry.data:
+        hardware_id = str(uuid.uuid4())
+        hass.config_entries.async_update_entry(
+            entry,
+            data={**entry.data, CONF_DEVICE_ID: hardware_id},
+        )
```

Comment:
Let's use a minor config entry migration for this

Reply:
Done